packages/lit-element/src/lit-element.ts

```diff
@@ -139,10 +130,14 @@ export class LitElement extends UpdatingElement {
    * the element to update.
    */
   protected render(): unknown {
-    return renderNotImplemented;
+    return noChange;
```

Comment:
Ah right, that deserves some discussion. @sorvell had suggested just re-implementing `LitElement`'s `update` in hydrate-support rather than adding a patchable `_renderImpl`, since LitElement's `update` is pretty small, and that would have required exporting the `renderNotImplemented` symbol. But in thinking about why we had that, it was to disambiguate whether the user implemented render so that we didn't clear the light DOM if the user had not implemented `render`. Since we don't clear the container now, I don't think we need to treat "not implemented" special anymore, we can just return `noChange` to not do anything in `render`. Seems better?

Reply:
FWIW, this was the change where `renderNotImplemented` was introduced: https://github.com/Polymer/lit-element/pull/917:
> https://github.com/Polymer/lit-element/issues/712 Introduced a breaking behavior change for situations where
> render is unimplemented, and DOM is added before being connected to
> the document.